### ISSUE TITLE
Wrong dates in day to day report

### ISSUE THREAD

**Issue by baswieb** [nightscout/cgm-remote-monitor#6853]
If I generate a day to day report in Nightscout (with date range "last two weeks" and sort order "newest on top") the first day shown is "monday 8-2-2021". The second day shown is "sunday 6-2-2021". This is wrong because the date of last sunday is 7-2-2021.

Nightscout version 14.1.0
Data is uploaded with Loop.
Timezone is "Europe/Amsterdam"
Everytime I generate a report I have to (re)set the timezone manually to "Europe/Amsterdam" in the profile editor.
When loop has uploaded the profile the Timezone in the profile becomes empty.

**Comment by baswieb** [comment]
![image](https://user-images.githubusercontent.com/8107422/107285825-02b8c400-6a60-11eb-9522-8a4fff0bf4b1.png)

**Comment by sulkaharo** [comment]
Right, so this sounds like a Loop issue, where it's uploading an incorrect profile. To get that fixed, you'd need to get in touch with the Loop developers.

**Comment by baswieb** [comment]
The timezone becoming empty can be a loop issue but the report showing "sunday 6-2-2021" when it should show "sunday 7-2-2021". Is this a bug in the report?

**Comment by Nightfoxy** [comment]
@sulkaharo  Is this the same issue?
When I run a day to day report for my daughter, the dates are correct.
When I run the day to day on someone else's site who is in a different time zone, the dates are off by 1.
![image](https://user-images.githubusercontent.com/46848705/120916130-9ee48780-c65c-11eb-9a39-113b58953829.png)

All involved are using Loop.

**Comment by AndyLow91** [comment]
Closing this as resolved.

The Day to Day report timezone/date handling has been fixed in `dev` through #8469 and #8485. Reports should now use the profile/report timezone consistently, which addresses the off-by-one-day/date-label problems described here.

If this is still reproducible on current `dev`, please comment with the profile timezone, selected report range, browser/server timezone if known, and a screenshot of the incorrect date labels.
